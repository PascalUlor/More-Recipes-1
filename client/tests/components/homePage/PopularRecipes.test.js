/* eslint-disable */
import React from 'react';
import { BrowserRouter as Router } from 'react-router-dom';
import toJson from 'enzyme-to-json';
import PopularRecipes from '../../../components/homePage/PopularRecipes.jsx';


let shallowWrapper;

describe('<PopularRecipes/>', () => {
  const initialProps = {
    fetchedPopularRecipes: []
  };
  it('renders popular recipes component without crashing', () => {
    shallowWrapper = shallow(<PopularRecipes {...initialProps}/>);
    expect(toJson(shallowWrapper)).toMatchSnapshot();
  });

  it('renders popular recipes component without crashing', () => {
    const newProps = {
      fetchedPopularRecipes: [{
        id: 1,
        recipeImage: 'recipeOne.jpg',
        title: '',
        User: { fullName: '' },
        upvotes: 1,
        downvotes: 0,
        viewsCount: 1
      }]
    };

    shallowWrapper = mount((
      <Router>
        <PopularRecipes {...newProps}/>
      </Router>
    ));
    expect(shallowWrapper.find('PopularRecipesList').length).toBe(1);
  });
});
