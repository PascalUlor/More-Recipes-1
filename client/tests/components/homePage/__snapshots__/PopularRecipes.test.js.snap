// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PopularRecipes/> renders popular recipes component without crashing 1`] = `
<section
  id="home"
>
  <div
    className="container"
  >
    <section
      className="cover text-center mb-3 mt-4"
    >
      <h3
        className="page-header"
      >
        Popular Recipes
      </h3>
      <hr
        id="favorites"
      />
    </section>
    <div
      className="lead text-center text-warning p-3"
    >
      There are no recipes to display
    </div>
    <div
      className="row"
    >
      <div
        className="col text-center"
      >
        <p
          className="lead font-weight-bold mb-5 mt-3"
        >
          Want to create your awesome recipe?
          <Link
            replace={false}
            to="/signup"
          >
             Sign Up Now
          </Link>
        </p>
      </div>
    </div>
  </div>
</section>
`;
